File: algorithm/fizzbuzz.rb
max = ARGV[0].to_i

(1..max).each do |num|
  if num % 15 == 0
    puts 'FizzBuzz'
  elsif num % 3 == 0
    puts 'Fizz'
  elsif num % 5 == 0
    puts 'Buzz'
  else
    puts num
  end
end

File: algorithm/fizzbuzz.go
package main

import (
	"flag"
	"fmt"
)

func main() {
	num := flag.Int("num", 0, "num")
	flag.Parse()

	var i int
	for i = 1; i <= *num; i++ {
		if i%15 == 0 {
			fmt.Println("FizzBuzz")
		} else if i%3 == 0 {
			fmt.Println("Fizz")
		} else if i%5 == 0 {
			fmt.Println("Buzz")
		} else {
			fmt.Println(i)
		}
	}
}
